src/rpc/dial.rs

```diff
@@ -550,21 +556,65 @@ impl DialBuilder<WithoutCredentials> {
         }
     }
 
+    async fn connect_mdns(self, original_uri: Parts) -> Result<ViamChannel> {
+        let mdns_uri = webrtc::action_with_timeout(self.get_mdns_uri(), Duration::from_millis(15))
+            .await
+            .ok()
+            .flatten()
+            .ok_or(anyhow::anyhow!(
+                "Unable to establish connection via mDNS; uri not found"
+            ))?;
+
+        self.connect_inner(Some(mdns_uri), original_uri).await
+    }
+
     pub async fn connect(self) -> Result<ViamChannel> {
         log::debug!("{}", log_prefixes::DIAL_ATTEMPT);
-        let original_uri = match self.duplicate_uri() {
-            Some(uri) => uri,
-            None => {
-                return Err(anyhow::anyhow!(
-                    "Attempting to connect but there was no uri"
-                ))
+        let original_uri = self.duplicate_uri().ok_or(anyhow::anyhow!(
+            "Attempting to connect but there was no uri"
+        ))?;
+        let original_uri2 = duplicate_uri(&original_uri).ok_or(anyhow::anyhow!(
+            "Attempting to connect but there was no uri"
+        ))?;
```

Comment:
Slightly annoying that this is duplicative. `original_uri2` will never fail (if it were to do so, `original_uri` would have failed before so we _could_ `unwrap()` rather than the `ok_or` block, but I think adding `unwrap`s is a bit uglier/less idiomatic than the minorly duplicative error handling here.